### PR TITLE
[5] Add Distance controller and tests

### PR DESCRIPTION
**Pull request by Yurchenko-Anton**
Create test persistence layer config and refactor old tests

### CODE REVIEW

File: src/main/java/com/example/matcher/dto/CreateOrderDTO.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package com.example.matcher.dto;
+
+import lombok.Value;
+
+@Value
+public class CreateOrderDTO {
+
+    String startPosition;
+
```
Comment (asinchuk) [inline src/main/java/com/example/matcher/dto/CreateOrderDTO.java:9]: please remove unnecessary lines

File: src/main/java/com/example/matcher/dto/LocationsDTO.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package com.example.matcher.dto;
+
+import lombok.Value;
+
+@Value
+public class LocationsDTO {
+
+    int id;
+
```
Comment (asinchuk) [inline src/main/java/com/example/matcher/dto/LocationsDTO.java:9]: same here

File: src/main/java/com/example/matcher/repository/DistanceRepository.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+                .from(LocationsEntity.INSTANCE)
+                .where(LocationsEntity.STREET_NAME.eq(locations))
+                .fetch();
+
+        int id = request.stream().findFirst().orElseThrow().get(LocationsEntity.ID);
+        String streetName = request.stream().findFirst().orElseThrow().get(LocationsEntity.STREET_NAME);
+        int xCoordinate = request.stream().findFirst().orElseThrow().get(LocationsEntity.X_COORDINATE);
+        int yCoordinate = request.stream().findFirst().orElseThrow().get(LocationsEntity.Y_COORDINATE);
+
+    return new LocationsDTO(id, streetName, xCoordinate, yCoordinate);
```
Comment (asinchuk) [inline src/main/java/com/example/matcher/repository/DistanceRepository.java:27]: please use lombok builder instead

File: src/main/java/com/example/matcher/service/DistanceService.java [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+    public Double getDistance(CreateOrderDTO createOrderDTO){
+        LocationsDTO startLocations = distanceRepository.getLocations(createOrderDTO.getStartPosition());
+        LocationsDTO finishLocations = distanceRepository.getLocations(createOrderDTO.getFinishPosition());
+
+        double distance = calculateService.calculateDistance(startLocations, finishLocations);
+
+        return distance;
+    }
+
```
Comment (asinchuk) [inline src/main/java/com/example/matcher/service/DistanceService.java:24]: redundant line

File: src/test/java/com/example/matcher/service/DistanceServiceTest.java [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+class DistanceServiceTest {
+
+    private final String START_POSITION = "Kosmonavtov";
+    private final String FINISH_POSITION = "Junosty";
+
+    private final TestPLConfig testPLConfig = new TestPLConfig();
```
Comment (asinchuk) [inline src/test/java/com/example/matcher/service/DistanceServiceTest.java:16]: private static final

File: src/test/java/com/example/matcher/service/DistanceServiceTest.java [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+    private final PLContext plContext = testPLConfig.configPL();
```
Comment (asinchuk) [inline src/test/java/com/example/matcher/service/DistanceServiceTest.java:17]: move to common this config

File: src/test/java/com/example/matcher/repository/LocationPersistenceTest.java [hunk 7]
```diff
@@ -17,17 +18,11 @@
...
-    private static final String USERNAME = "root";
-    private static final String PASSWORD = "root";
-
     private static final String START_STREET_NAME = "Test";
     private static final String UPDATE_STREET_NAME = "UpdateTest";
 
-    private static final DSLContext dslContext = DSL.using(URL, USERNAME, PASSWORD);
-
-    private static final PLContext plContext = new PLContext.Builder(dslContext).withAuditRecordPublisher(new JsonAuditRecordPublisher()).build();
+    private static final TestPLConfig testPLConfig = new TestPLConfig();
```
Comment (asinchuk) [inline src/test/java/com/example/matcher/repository/LocationPersistenceTest.java:24]: move to common test base class pl config and extend test classes from it, no need to create same entity multiple times

File: src/main/java/com/example/matcher/dto/CreateOrderDTO.java [hunk 8]
```diff
@@ -4,10 +4,7 @@
 
 @Value
 public class CreateOrderDTO {
-
     String startPosition;
-
     String finishPosition;
-
     Long guestId;
```
Comment (asinchuk) [inline src/main/java/com/example/matcher/dto/CreateOrderDTO.java:9]: userId?

**Review by asinchuk — APPROVED** [review]

File: src/main/java/com/example/matcher/dto/CreateOrderDTO.java (hunk 8 above)
Comment (Yurchenko-Anton) [reply src/main/java/com/example/matcher/dto/CreateOrderDTO.java:9]: if the user orders a taxi not for himself

Comment (asinchuk) [reply src/main/java/com/example/matcher/dto/CreateOrderDTO.java:9]: Guest should be registered at system? Maybe we should have mechanism of the notifications for users that use our services but not our customers? Also we can have more then one additional user.

**Pull request merged**